File: app/src/main/java/com/example/inventory/data/Item.kt
package com.example.inventory.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.sql.Date

/**
 * thông tin hóa đơn của user được lưu tại đây
 */
@Entity(
    tableName = "items",
    foreignKeys = [
        ForeignKey(
            entity = Receipt::class,
            parentColumns = ["receiptId"],
            childColumns = ["receiptId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Item(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val name: String,
    val price: Double,
    @ColumnInfo(name = "quantity")
    val quantity: Float,
    val date: Date,
    val store: String,
    val category: String,
    @ColumnInfo(name = "receiptId")
    val receiptId: Int,
    val calories: Double? = null,
    val protein: Double? = null,
    val carbs: Double? = null,
    val fats: Double? = null,
)

File: api/schemas/item.py
from pydantic import BaseModel
from datetime import datetime

class Item(BaseModel):
    id: int | None = 0
    name: str | None = None
    price: float | None = None
    quantity: float | None = None
    date: datetime | None = None
    store: str | None = None
    category: str | None = None
    receiptId: int | None = None
    protein: float | None = None
    carbs: float | None = None
    fats: float | None = None
    calories: float | None = None
